src/Components/OATPropertyEditor/Internal/PropertyList/PropertyList.tsx

```diff
@@ -170,9 +183,25 @@ const PropertyList: React.FC<IPropertyListProps> = (props) => {
                     property.name
                 );
                 updatedContents[index].name = args.name;
-                updateModel({
+                const updatedModel = {
                     ...selectedItem,
                     contents: updatedContents
+                };
+
+                // validate changes
+                validateOntology({
+                    existingModels: oatPageState.currentOntologyModels,
+                    selectedModelId: oatPageState.selection?.modelId,
+                    originalItem: selectedItem,
+                    updatedItem: updatedModel
+                }).then((validationResult) => {
```

Comment:
should you also have `.catch` block for this promise to call `showErrorDialog(error);` in case validateOntology fails?

Reply:
hmmm. fair enough. I can add that.